narwhals/functions.py

```diff
@@ -430,6 +432,163 @@ def _from_dict_impl(
     return from_native(native_frame, eager_only=True)
 
 
+def from_numpy(
+    data: np.ndarray,
+    schema: dict[str, DType] | Schema | None = None,
```

Comment:
I think it would be nice to support `list[str]` as column names. In my opinion is way more common to specify the names of the columns without their dtypes when creating a dataframe

Reply:
Agreed, I added this as an option for `schema`